packages/components/src/Filters/FilterInput.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from 'react';
+import { Checkbox, Radio } from '@patternfly/react-core';
+
+export interface FilterInputProps {
+  addRemoveFilters?: (value?: string, param?: string, type?: string, checked?: boolean) => void;
+  className?: string;
+  currentPage?: string;
+  id: string;
+  label: React.ReactNode;
+  param: string;
+  value?: string;
+  filters?: Record<string, string>;
+  type?: string;
+  onChange?: (checked: boolean, event: React.FormEvent<HTMLInputElement>) => void;
+}
+
+const FilterInput: React.FC<FilterInputProps> = ({ addRemoveFilters, param, filters, type = 'checkbox', value, id, label }) => {
+  const [checked, setChecked] = useState(false);
+  useEffect(() => {
+    if (type !== 'radio') {
+      if (filters && value && param in filters && filters[param] && filters[param].includes(value)) {
+        setChecked(true);
+      } else {
+        setChecked(false);
+      }
+    }
+  });
+  const handleChange = () => {
+    switch (type) {
+      case 'checkbox':
+        setChecked(!checked);
+        break;
+    }
+    addRemoveFilters && addRemoveFilters(value, param, type, checked);
+  };
+  if (type !== 'radio') {
+    return <Checkbox aria-label={param} id={id} isChecked={checked} label={label} onChange={handleChange} value={value} ouiaId={id} />;
```

Comment:
Preivously the `aria-label` was `label` any reason to switch it to `param`? Also the `ouiaId` was previously `label` that would break many, many tests.
```suggestion
    return <Checkbox aria-label={label} id={id} isChecked={checked} label={label} onChange={handleChange} value={value} ouiaId={label} />;
```

Reply:
label PropType is node and aria-label is supposed to be a string

Reply:
The other one I changed it back

Reply:
Good point about the `label` being a node vs `string`. I'd just bluntly cast the value to `string`. We can investigate further the update to use `param` instead of `label, but I'd like to avoid API changes as much as possible.